### ISSUE TITLE
Disable conan checking compiler comes after compiler if statement in conanbuildinfo

### ISSUE THREAD

**Issue by trondhe** [conan-io/conan#4268]
In the conanbuildinfo.cmake file created in the build folder, the function `conan_check_compiler` the check for `CONAN_DISABLE_CHECK_COMPILER `comes after a check for `if(NOT DEFINED CMAKE_CXX_COMPILER_ID)` as shown in this extract:
``` cmake
function(conan_check_compiler)
    if(NOT DEFINED CMAKE_CXX_COMPILER_ID)
        if(DEFINED CMAKE_C_COMPILER_ID)
            message(STATUS "This project seems to be plain C, using '${CMAKE_C_COMPILER_ID}' compiler")
            set(CMAKE_CXX_COMPILER_ID ${CMAKE_C_COMPILER_ID})
            set(CMAKE_CXX_COMPILER_VERSION ${CMAKE_C_COMPILER_VERSION})
        else()
            message(FATAL_ERROR "This project seems to be plain C, but no compiler defined")
        endif()
    endif()
    if(CONAN_DISABLE_CHECK_COMPILER)
        message(STATUS "WARN: Disabled conan compiler checks")
        return()
    endif()
    ...
```

In CMake setting `project(myproj LANGUAGE NONE)` results in this evaluating to FATAL_ERROR regardless if `CONAN_DISABLE_CHECK_COMPILER` is defined or not.

Is there something I'm overlooking or is it a bug?

**Comment by memsharded** [comment]
The idea was that the compiler should be there, and CMake should have either a CXX or a C compiler to build the project.

But yes, I think that the disabling of the check should have higher priority, and it shouldn't fail for those scenarios that the project language is not defined, lets try to change it.

**Comment by memsharded** [comment]
It has been fixed, will be released in next conan 1.12 (or test today in the https://test.pypi.org/project/conan/). Thanks for reporting!
